src/modules/UI/scenes/CreateWallet/CreateWalletName.ui.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import React, {Component} from 'react'
+import {Actions} from 'react-native-router-flux'
+import {
+  ActivityIndicator,
+  Alert,
+  View,
+  Keyboard
+} from 'react-native'
+import {SecondaryButton, PrimaryButton} from '../../components/Buttons'
+import {FormField} from '../../../../components/FormField.js'
+
+import styles from './style.js'
+import {MaterialInputOnWhite} from '../../../../styles/components/FormFieldStyles.js'
+import s from '../../../../locales/strings.js'
+import Gradient from '../../components/Gradient/Gradient.ui'
+
+const WALLET_NAME_INPUT_PLACEHOLDER = s.strings.fragment_wallets_addwallet_name_hint
+const CANCEL_TEXT = s.strings.string_cancel_cap
+const WALLET_NAME_INVALID_TEXT = s.strings.create_wallet_invalid_name
+const WALLET_NAME_ENTER_VALID_TEXT = s.strings.create_wallet_enter_valid_name
+const INVALID_DATA_TEXT = s.strings.fragment_create_wallet_select_valid
+const NEXT_TEXT = s.strings.string_next_capitalized
+
+export type Props = {
+
+}
+
+export type State = {
+  walletName: string
+}
+
+export class CreateWalletName extends Component<Props, State> {
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      walletName: ''
+    }
+  }
+
+  isValidWalletName = (): boolean => {
+    const {walletName} = this.state
+    const isValid: boolean = walletName.length > 0
+
+    return isValid
+  }
+
+  onNext = (): void => {
+    if (this.isValidWalletName()) {
+      Actions.createWalletSelectCrypto({walletName: this.state.walletName})
```

Comment:
Going forward. Avoid directly using the scene name in an Action, use the constant instead as this can be checked with eslint.

`Actions[Constants.CREATE_WALLET_SELECT_CRYPTO]`